lib/chef/azure/commands/enable.rb

```diff
@@ -10,7 +10,18 @@
 require 'tempfile'
 require 'chef/azure/core/windows_bootstrap_context'
 require 'erubis'
-require 'chef/knife'
+# Using knife from separately extracted knife gem
+include Chef::Mixin::ShellOut
+ver = shell_out("chef-client -v").stdout
+arr = ver.split(': ')
+if arr[1].start_with?("15") || arr[1].start_with?("16")
+  require 'chef/knife/core/bootstrap_context'
+else
+  find = RUBY_PLATFORM =~ /mswin|mingw|windows/ ? Dir["C:/opscode/chef/knife/*"] : Dir["/opt/chef/knife/*"]
+  path1 = find.join('/')
+  path2 = path1 + '/lib/chef/knife'

```

Comment:
Can it be something like 
`path1 = find.join('/') + '/lib/chef/knife'`
`require path1`